state_synchronizer/src/peer_manager.rs

```diff
@@ -68,16 +82,43 @@ impl PeerManager {
         self.get_active_upstream_peer_ids().is_empty()
     }
 
+    pub fn update_score(&mut self, peer_id: &PeerId, update_type: PeerScoreUpdateType) {
+        if let Some(peer_info) = self.peers.get_mut(peer_id) {
+            match update_type {
+                PeerScoreUpdateType::Success => {
+                    let current_score = peer_info.score;
+                    peer_info.score = current_score.max(MAX_SCORE);
+                }
+                PeerScoreUpdateType::InvalidChunk => {
+                    peer_info.score *= 0.8;
```

Comment:
this also has potential issue that if you decrease score too often, it will be eventually rounded down to zero, and you may end up with no peers to sync